### PR TITLE
User search

### PR DESCRIPTION
**Pull request by minhnghia2208**
Changes:
/users/search?firstName=[string]&lastName=[string]     // Must specify both
/users/search?search=[string]      // search for both firstname and lastname

recognitions is similar

### CODE REVIEW

**Review by dima618 — APPROVED** [review]

**Review by dima618 — CHANGES_REQUESTED** [review]
I'm getting some query errors for some reason, so I'm going to request changes for now until I see why it's doing it.

EDIT: Query issues were on my end, but I requested some other specific changes

File: src/users/users.service.ts [hunk 1]
```diff
@@ -234,48 +234,18 @@ export class UsersService {
...
-        ]);
-        // queryBuilder.orWhere({lastName: Like(`%${lastName}%`)});
+        queryBuilder.orderBy('user.firstName', 'ASC')
+        // Must specify both firstname and lastname
+        .where("user.firstName ilike :firstName", {firstName: '%'+firstName+'%'})
+        .andWhere("user.lastName ilike :lastName", {lastName: '%'+lastName+'%'})
+
+        // search: string return users with similar firstname and lastname
+        .orWhere("user.firstName ilike :search", {search: '%'+search+'%'})
+        .orWhere("user.lastName ilike :search", {search: '%'+search+'%'});
```
Comment (dima618) [inline src/users/users.service.ts]: Can we also make this check for if firstname + lastname is similar to search and firstname + " " + lastname is similar to search?

File: src/users/users.service.ts [hunk 2]
```diff
@@ -234,48 +234,18 @@ export class UsersService {
...
-    async paginate_username(options: IPaginationOptions, firstName: string, lastName: string): Promise<Pagination<Users>> {
+   
+    async paginate_username(options: IPaginationOptions, firstName: string, lastName: string, search: string): Promise<Pagination<Users>> {
         const queryBuilder = this.usersRepository.createQueryBuilder('user');
-        queryBuilder.where([
-            {firstName: Like(`%${firstName}%`)},
-            {lastName: Like(`%${lastName}%`)}
-        ]);
-        // queryBuilder.orWhere({lastName: Like(`%${lastName}%`)});
+        queryBuilder.orderBy('user.firstName', 'ASC')
```
Comment (dima618) [inline src/users/users.service.ts:241]: We should also be only looking for users with the same companyId as the JWT token

File: src/users/users.service.ts (hunk 1 above)
Comment (dima618) [reply src/users/users.service.ts]: Also, do we wan't to exclude the JWT user from the search results?

**Review by dima618 — CHANGES_REQUESTED** [review]

File: src/users/users.service.ts (hunk 1 above)
Comment (minhnghia2208) [reply src/users/users.service.ts]: > Also, do we wan't to exclude the JWT user from the search results?

Aside from me being lazy, I think seeing my own profile from others user perspective is a neat feature.

Comment (minhnghia2208) [reply src/users/users.service.ts]: > Can we also make this check for if firstname + lastname is similar to search and firstname + " " + lastname is similar to search?

Fixed

**Pull request closed without merging**